tests/5.1/metadirective/test_metadirective_target_device_kind.c

```diff
@@ -0,0 +1,63 @@
+//===--------------------- test_metadirective_target_device_kind.c ---------------------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+// 
+// Tests that target_device is recognized & will work properly when matching an nvidia
+// GPU. More specifically, uses the kind clause to see if gpu or nohost is recognized
+// as the target device's "kind" implementation. 
+//
+////===---------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int test_metadirective_target_device() {
+   int errors = 0;
+   int A[N];
+
+   for (int i = 0; i < N; i++) {
+      A[i] = 1;
+   }
+
+   #pragma omp target map(tofrom: A)
+   {
+      #pragma omp parallel num_threads(4)
```

Comment:
Remove num_threads or try num_threads(128) (similar problem to PR #656), test with GCC & NVC